pandas/tests/util/test_show_versions.py

```diff
@@ -26,11 +37,44 @@
     "ignore:Distutils:UserWarning"
 )
 @pytest.mark.filterwarnings("ignore:Setuptools is replacing distutils:UserWarning")
-def test_show_versions(capsys):
+def test_show_versions(tmpdir):
+    as_json = os.path.join(tmpdir, "test_output.json")
+
+    pd.show_versions(as_json=as_json)
+
+    with open(as_json) as fd:
+        # check if file output is valid JSON, will raise an exception if not
+        result = json.load(fd)
+
+    # Basic check that each version element is found in output
+    expected = {
+        "system": _get_sys_info(),
+        "dependencies": _get_dependency_info(),
+    }
+
+    assert result == expected
+
+
+def test_show_versions_console_json(capsys):
+    pd.show_versions(as_json=True)
+    result = get_captured_output(capsys)
+
+    # check valid json is printed to the console if as_json is True
+    result = json.loads(result)
```

Comment:
instead of reassigning to `result` (which may cause problems when we expand mypy checks on tests) could you just do something like
```
stdout = capsys.readouterr().out
result = json.loads(stdout)
```
I'm not sure that making a function for `get_captured_output` adds much, repetition is more acceptable in tests than it is in source code